apis/users.py

```diff
@@ -291,6 +292,38 @@ def change_password():
 
 
 
+@user_api.post("/change_username")
+@check_user
+def change_username():
+    '''
+    Given a new username, change the username
+    '''
+    update_query = "UPDATE `users` SET `username`=%s WHERE `user_id`=%s"
+
+    if ("new_username" not in request.json):
+        #print("Incomplete request")
+        return ("Incomplete request", 400)
+
+    id = session["user_id"]
+    new_username = request.json["new_username"]
+
+    db = get_db()
+    with db.cursor(cursor=DictCursor) as cursor:
+        
+        try:
+            cursor.execute(update_query, (new_username, id))
+            db.commit()
+        
+        except pymysql.IntegrityError:
+            print("dup username")
+            return (f'Username `{new_username}` already exists', 409)
+        except pymysql.Error as e:
+            return ("Unknown error", 500)
+
```

Comment:
You should probably update the session here, otherwise the header won't update I think

Reply:
Added